netket/sampler/metropolis.py

```diff
@@ -54,8 +54,10 @@ class MetropolisSamplerState(SamplerState):
         default_factory=lambda: jnp.zeros((), dtype=jnp.int64)
     )
     """Number of moves performed along the chains in this process since the last reset."""
-    n_accepted_proc: int = struct.field(
-        default_factory=lambda: jnp.zeros((), dtype=jnp.int64)
+    n_accepted_proc: jnp.ndarray = struct.field(
+        default_factory=lambda: jnp.zeros(
+            (), dtype=jnp.int64
+        )  # TODO (c): initialize with shape σ.shape[0] to avoid recompilation
```

Comment:
I assume I would have to use __post__init__ to infer the shape from \sigma?

Reply:
yes. Either you initialise it with `None` and then set it to whatever you need,
or you use `pre_init` and add it to the kwargs.